Change expected error code in Python negative interop client

ncteisen opened the pull request:
The test now expects to receive StatusCode.INTERNAL when it receives a
RST_STREAM from the server.

This is a subsequent fix following from #9142, which was a bug in the c core that caused RST_STREAM(0) to fail with UNKNOWN instead of INTERNAL

ncteisen commented:
cc @makdharma

nathanielmanistaatgoogle commented:
Change content looks good, [commit message could be tweaked](http://chris.beams.io/posts/git-commit/#seven-rules), but I think I might want to hold this up in case it conflicts with [pull request 9276](https://github.com/grpc/grpc/pull/9276).

ncteisen commented:
I also just added a call to grpc.channel_ready_future, because I was seeing some flakiness when I tested this.

Do you have an ETA on the pyformat PR? I can wait or I can just ensure this file is up to the standards.

ncteisen commented:
Just updated this to comply with the recent PR enforcing good python formatting. Hopefully I did it correctly and Jenkins is happy.

Also, I realized I never removed the prepended underscore from the client file. In a previous PR (#9097) we decided that since it was not a unit test, it did not deserve to have a leading underscore. I renamed the file in the PR

nathanielmanistaatgoogle commented:
Poke me over lunch one day and I'll share the two competing philosophies I have about leading underscore on test module names. Let's be ginger about enforcing either one until I actually commit to one or the other in writing.

ncteisen commented:
It was my commit message. Updated it to link to statuscodes.md

nathanielmanistaatgoogle commented:
Which row of the table applies? "Flow-control protocol violation"?

ncteisen commented:
Sorry, I was moving too fast there. https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md is the applicable doc in this situation. 

Specifically the RST_STREAM table, with the NO_ERROR(0) row.

ncteisen commented:
Only failure is a well known flake #9093. Merging
